sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -78,8 +82,23 @@ case class AddJar(path: String) extends RunnableCommand {
   override def run(sqlContext: SQLContext) = {
     val hiveContext = sqlContext.asInstanceOf[HiveContext]
     hiveContext.runSqlHive(s"ADD JAR $path")
-    hiveContext.sparkContext.addJar(path)
-    Seq.empty[Row]
+    val substituted =
+      new VariableSubstitution().substitute(sqlContext.asInstanceOf[HiveContext].hiveconf, path)
+    hiveContext.sparkContext.addJar(substituted)
```

Comment:
As I remember, every SQL text is substitute in https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala#L86
Can you confirm if we really need a variable substitution here?